packages/bee-q/src/components/option/bq-option.tsx

```diff
@@ -0,0 +1,170 @@
+
+  /** Handler to be called when item is clicked */
+  @Event() bqOptionClick: EventEmitter<HTMLBqOptionElement>;
+
+  /** Handler to be called on enter key press */
+  @Event() bqOptionEnter: EventEmitter<HTMLBqOptionElement>;
+
+  /** Handler to be called on enter key up */
+  @Event() bqOnEnterKeyUp: EventEmitter<HTMLBqOptionElement>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  // Listeners
+  // ==============
+
+  @Listen('keydown')
+  onKeyDown(event: KeyboardEvent) {
+    if (event.key === 'Enter') this.bqOptionEnter.emit(this.el);
+  }
+
+  @Listen('keyup')
+  onKeyUp(event: KeyboardEvent) {
+    if (event.key === 'Enter') this.bqOnEnterKeyUp.emit(this.el);
+  }
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private onBlur = (event: Event) => {
+    if (this.disabled) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    this.bqOptionBlur.emit(this.el);
+  };
+
+  private onFocus = (event: Event) => {
+    if (this.disabled) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    this.bqOptionFocus.emit(this.el);
+  };
+
+  private onClick = (event: Event) => {
+    if (this.disabled) {
+      event.preventDefault();
+      event.stopPropagation();
+      return;
+    }
+
+    this.bqOptionClick.emit(this.el);
+  };
+
+  private onSlotChange = () => {
+    this.hasPrefix = hasSlotContent(this.prefixElem, 'prefix');
+  };
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <div
+        class={{
+          'bq-option': true,
+          disabled: this.disabled,
+          active: !this.disabled && this.selected,
+        }}
+        role="option"
+        aria-selected={this.selected}
+        aria-disabled={this.disabled}
+        tabindex={this.disabled ? '-1' : '0'}
+        onBlur={this.onBlur}
+        onFocus={this.onFocus}
+        onClick={this.onClick}
+        part="base"
+      >
+        <span class="bq-option__child" ref={(elem) => (this.prefixElem = elem)} part="prefix">
+          <slot name="prefix" onSlotchange={this.onSlotChange} />
+        </span>
+        <span
+          class={{
+            'bq-option__child': true,
+            label: true,
+            'no-prefix': !this.hasPrefix,
+          }}
+          part="label"
+        >
+          <slot />
+        </span>
+        <span class="bq-option__child suffix" part="suffix">
+          <slot name="suffix" />
+        </span>
```

Comment:
I just noticed that these 3 elements have the same CSS class name, why? 🤔

Reply:
Also, the prefix and suffix disappear when any of the prop changes in the component

https://github.com/Endava/bee-q/assets/328492/77d6ef98-97b5-4b45-a99b-4694fb0d2561

Reply:
The way you set the prefix and suffix element in the option storybook template is the culprit of the issue above ☝🏼. Avoid to over-engineer stuff, the simplest way will always be the best 😉 

```ts
// bq-options.stories.ts: #line 47

const Template = (args: Args) => html`
  <bq-option-list>
    <bq-option
      ?disabled=${args.disabled}
      ?selected=${args.selected}
      @bqOptionBlur=${args.bqOptionBlur}
      @bqOptionFocus=${args.bqOptionFocus}
      @bqOptionClick=${args.bqOptionClick}
    >
      <bq-icon name="user" slot="prefix"></bq-icon>
      <span>${args.text}</span>
      <bq-icon name="gear" slot="suffix"></bq-icon>
    </bq-option>
  </bq-option-list>
`;
```

Reply:
Actually here the intention was to have only 1 template for the stories. So, there are 3 stories (`Default`, `Active` and `Disabled`). For `Disabled` story, I wanted to have a different icons displayed in `prefix` and `suffix` slot. So, I needed to configure the template based on `args`. This way we would had only 1 template for all and not duplicate template based on 2 nodes.
I don't have a better solution than to have a separate template for `Disabled` story. (or maybe not to change the icons at all, and then no need for a separate template).

Reply:
You can have a hidden `iconName` argument, and `showIcon`, and based on that you will show the `bq-icon` or not ([see this example here](https://github.com/Endava/bee-q/blob/develop/packages/bee-q/src/components/input/_storybook/bq-input.stories.tsx))... but the current solution is not stable since when any property changes, the element object you're passing down is removed.